sdks/go/pkg/beam/core/runtime/exec/fn.go

```diff
@@ -224,14 +227,17 @@ func (n *invoker) Invoke(ctx context.Context, pn typex.PaneInfo, ws []typex.Wind
 }
 
 // ret1 handles processing of a single return value.
-// Errors or single values are the only options.
+// Errors, single values, or a ProcessContinuation are the only options.
 func (n *invoker) ret1(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime, r0 interface{}) (*FullValue, error) {
 	switch {
 	case n.outErrIdx >= 0:
 		if r0 != nil {
 			return nil, r0.(error)
 		}
 		return nil, nil
+	case n.outPcIdx >= 0:
+		n.ret = FullValue{Windows: ws, Timestamp: ts, Pane: pn, Continuation: r0.(sdf.ProcessContinuation)}
```

Comment:
Interesting question. We've established that we want an explicit stop value from the DoFn in the interface, so we'd probably not want to treat a nil value as a stop. A nil value here isn't technically problematic since we'd ignore it when we pick up the FullValue again and treat it as if the process completed successfully, but you're right in that we shouldn't allow users to do that

Reply:
I dropped in the panics into ret1 and ret2 to start, may add them to ret3+ if we're happy with how it looks.

Reply:
Left a comment about the specific string, but I think that is a good thing and I'd vote we expand to other rets

Reply:
Fixed (there's no Fn.Name() field so I opted just to use the string output for the Fn object)

Reply:
Ah, I was thinking of n.Fn.Fn.Name() 🙃 using the full object works for me